sys/net/application_layer/asymcute/asymcute.c

```diff
@@ -490,6 +497,10 @@ static void _on_puback(asymcute_con_t *con, const uint8_t *data, size_t len)
                                           MQTTSN_PUBLISH);
     if (req == NULL) {
         mutex_unlock(&con->lock);
+        /* Acknowledge due to an invalid QoS level 0 packet? */
+        if((data[6] != MQTTSN_ACCEPTED) && (data[4] == 0) && (data[5]==0)) {
```

Comment:
```suggestion
        if ((data[6] != MQTTSN_ACCEPTED) && (data[4] == 0) && (data[5]==0)) {
```

Also, this is unrelated to the introduced `mutex_unlock` call, could you elaborate on why this is added?

Reply:
The MQTT-SN specification states in 5.4.13 that a PUBACK is not only sent in the GOOD case at QoS levels 1 and 2 but also in the event of an error (It can also be sent as response to a PUBLISH message in case of an error; the error reason is then indicated in the ReturnCode field). The fact that these are errors for QoS level 0 can be seen from the MsgID description of PUBLISH (otherwise coded 0x0000).
Hence the missing query